go/billing/tests/test_api.py

```diff
@@ -259,6 +259,110 @@ def test_low_credit_notification_zero_last_topup_value(self):
+            transaction_type=Transaction.TRANSACTION_TYPE_MESSAGE)
+
+        self.assertFalse(transaction1['credit_cutoff_reached'])
+        self.assertEqual(Transaction.objects.count(), 2)
+
+    @inlineCallbacks
+    def test_credit_cutoff_inbound_reached(self):
+        self.patch(app_settings, 'ENABLE_LOW_CREDIT_CUTOFF', True)
+
+        mk_message_cost(
+            tag_pool=self.pool1,
+            message_direction=MessageCost.DIRECTION_INBOUND,
+            message_cost=0.4,
+            )
+
+        load_account_credits(self.account, 10)
+
+        transaction_kwargs = {
+            'account_number': self.account.account_number,
+            'message_id': 'msg-id-1',
+            'tag_pool_name': 'pool1',
+            'tag_name': 'tag1',
+            'message_direction': MessageCost.DIRECTION_INBOUND,
+            'session_created': False,
+            'transaction_type': Transaction.TRANSACTION_TYPE_MESSAGE
+        }
+
+        yield self.create_api_transaction(**transaction_kwargs)
+        transaction = yield self.create_api_transaction(**transaction_kwargs)
+
+        self.assertTrue(transaction['credit_cutoff_reached'])
+        self.assertEqual(Transaction.objects.count(), 3)
+
+    @inlineCallbacks
+    def test_credit_cutoff_outbound_not_reached(self):
+        self.patch(app_settings, 'ENABLE_LOW_CREDIT_CUTOFF', True)
+
+        mk_message_cost(
+            tag_pool=self.pool1,
+            message_direction=MessageCost.DIRECTION_OUTBOUND,
+            message_cost=0.2,
+            )
+
+        load_account_credits(self.account, 10)
+
+        transaction1 = yield self.create_api_transaction(
+            account_number=self.account.account_number,
+            message_id='msg-id-1',
+            tag_pool_name='pool1',
+            tag_name='tag1',
+            message_direction=MessageCost.DIRECTION_OUTBOUND,
+            session_created=False,
+            transaction_type=Transaction.TRANSACTION_TYPE_MESSAGE)
+
+        self.assertFalse(transaction1['credit_cutoff_reached'])
+        self.assertEqual(Transaction.objects.count(), 2)
+
+    @inlineCallbacks
+    def test_credit_cutoff_outbound_reached(self):
+        self.patch(app_settings, 'ENABLE_LOW_CREDIT_CUTOFF', True)
+
+        mk_message_cost(
+            tag_pool=self.pool1,
+            message_direction=MessageCost.DIRECTION_OUTBOUND,
+            message_cost=0.4,
+            )
+
+        load_account_credits(self.account, 10)
+
+        transaction_kwargs = {
+            'account_number': self.account.account_number,
+            'message_id': 'msg-id-1',
+            'tag_pool_name': 'pool1',
+            'tag_name': 'tag1',
+            'message_direction': MessageCost.DIRECTION_OUTBOUND,
+            'session_created': False,
+            'transaction_type': Transaction.TRANSACTION_TYPE_MESSAGE
+        }
+
+        yield self.create_api_transaction(**transaction_kwargs)
+        transaction = yield self.create_api_transaction(**transaction_kwargs)
```

Comment:
Same suggestions about loading fewer credits and checking the returned transaction here.